src/platform/Darwin/ConfigurationManagerImpl.cpp

```diff
@@ -137,14 +138,55 @@ ConfigurationManagerImpl & ConfigurationManagerImpl::GetDefaultInstance()
 
 CHIP_ERROR ConfigurationManagerImpl::Init()
 {
-    CHIP_ERROR err;
-
+#if CHIP_DEVICE_CONFIG_ENABLE_PERSISTENT_STORAGE
     // Initialize the generic implementation base class.
-    err = Internal::GenericConfigurationManagerImpl<PosixConfig>::Init();
-    SuccessOrExit(err);
+    ReturnErrorOnFailure(Internal::GenericConfigurationManagerImpl<PosixConfig>::Init());
 
-exit:
-    return err;
+    ReturnErrorOnFailure(DeviceLayer::PersistedStorage::KeyValueStoreMgrImpl().Init("chip.store"));
```

Comment:
Please let's be careful not to introduce more of the non-hermetic stuff we already need to fix on Linux with a single filename. Is there a way to make this configurable all the way from the command line?

Reply:
I agree that we need to have a way to configure this filename. In fact it is on my way of getting YAML tests against multiple accessories so I at the end of the day I will fix it.
That said `chip.store` is not introduced by this PR, I have just moved this call from `Server.cpp` to happens sooner , otherwise the KVS init on Darwin was happening after the stack initialisation and that was causing issues.
So, while I will fix it, that's not for this PR :)